src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -93,47 +121,86 @@ void Bank::evaluate_and_allocate() {
         }
         storage_guard.unlock();
 
-        ov::parallel_for(vec.size(), [&](std::size_t idx) {
-            const auto& lt = vec[idx];
+        std::vector<AllocatedTensor> uids_to_allocated(uid_count);
+
+        // No allocation needed
+        if (device_for_alloc == "CPU") {
+            ov::parallel_for(vec.size(), [&](std::size_t idx) {
+                const auto& lt = vec[idx];
+                std::unique_lock dev_guard(device_bank.mutex);
+                auto iter_device_registered = device_bank.registered_tensors.find(lt);
+                NPUW_ASSERT(iter_device_registered != device_bank.registered_tensors.end() &&
+                            "Tensor should be registered first!");
+                auto uid = iter_device_registered->second;
+                if (device_bank.storage[uid].tensor) {
+                    // Already allocated
+                    return;
+                }
+
+                dev_guard.unlock();
+                auto transformed = lt.eval();
+
+                // Note: No need to relock here as the storage is already prepared for writing
+                device_bank.storage.at(uid).tensor = std::move(transformed);
```

Comment:
do you have something to do with `transformed`? it seems it just could be written as `.tensor = lt.eval()`  ?

Reply:
Done